From fa417c369b48a8f7acaf9aba1ff94b0969b0cb7e Mon Sep 17 00:00:00 2001
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Date: Thu, 3 Mar 2011 21:10:05 +0300
Subject: USB: OHCI: use pci_dev->revision
Patch-mainline: v2.6.39
Git-commit: fa417c369b48a8f7acaf9aba1ff94b0969b0cb7e

Commit ab1666c1364a209e6141d7c14e47a42b5f00eca2 (USB: quirk PLL power down mode)
added code that reads the revision ID from the PCI configuration register while
it's stored by PCI subsystem in the 'revision' field of 'struct pci_dev'...

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/host/ohci-pci.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/usb/host/ohci-pci.c
+++ b/drivers/usb/host/ohci-pci.c
@@ -151,7 +151,7 @@ static int ohci_quirk_amd700(struct usb_
 {
 	struct ohci_hcd *ohci = hcd_to_ohci(hcd);
 	struct pci_dev *amd_smbus_dev;
-	u8 rev = 0;
+	u8 rev;
 
 	if (usb_amd_find_chipset_info())
 		ohci->flags |= OHCI_QUIRK_AMD_PLL;
@@ -161,7 +161,7 @@ static int ohci_quirk_amd700(struct usb_
 	if (!amd_smbus_dev)
 		return 0;
 
-	pci_read_config_byte(amd_smbus_dev, PCI_REVISION_ID, &rev);
+	rev = amd_smbus_dev->revision;
 
 	/* SB800 needs pre-fetch fix */
 	if ((rev >= 0x40) && (rev <= 0x4f)) {
